sdk/cosmos/azure-cosmos/pom.xml

```diff
@@ -280,18 +280,6 @@ Licensed under the MIT License.
         </configuration>
       </plugin>
 
-      <!--  CosmosSkip - Needed temporary to not fail on warning on compilation -->
```

Comment:
Do we need to get rid of the whole plugin ?
Why can't we just flip the `failOnWarning` flag to true. 
In my understanding, by this plugin -> we decide on the java compiler version used for our SDK. 
In my opinion, we should keep this.

Reply:
@kushagraThapar - we removed it to inherit the compiler plugin and related settings from the parent pom file. I'm not sure any of the advantages we get by having it in locally, @JonathanGiles thoughts?

Reply:
I see, yeah inheriting should be totally fine, didn't realize the parent pom's plugin. I am good with this change.